components/org.apache.stratos.rest.endpoint/src/main/java/org/apache/stratos/rest/endpoint/api/StratosApiV41.java

```diff
@@ -441,7 +441,7 @@ public Response removeCartridge(
     @AuthorizationAction("/permission/protected/manage/addServiceGroup")
     @SuperTenantService(true)
     public Response addServiceGroup(
-            GroupBean serviceGroupDefinition) throws RestAPIException {
+            GroupBean serviceGroupDefinition) throws RestAPIException, InvalidCartridgeGroupDefinitionException {
```

Comment:
Is it required to throw InvalidCartridgeGroupDefinitionException from this API method? If so there should be an exception handler which captures it and generate an error response to the client. IMO API methods should only throw RestAPIException.